include/dao/db_manager.js

```diff
@@ -185,6 +199,91 @@ module.exports = function DBManagerModule(pb) {
                 };
                 cb(err, result);
             });
+        }
+
+
+        /**
+         * Sorts through all created indices and drops any index not declared in pb.config
+         * @method dropUnconfiguredIndices
+         * @param {Array} procedures
+         * @param {Function} cb
+         */
+        this.dropUnconfiguredIndices = function(procedures, cb) {
+            var self = this;
+            //to prevent a cirular dependency we do the require for DAO here.
+            var DAO = require('./dao.js')(pb);
+            this.getStoredIndices(function(err, storedIndices) {
```

Comment:
Same change made here.  Used a callback with  return instead.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Sunday, July 12, 2015 at 7:04 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In include/dao/db_manager.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34426759>:

> @@ -185,6 +199,91 @@ module.exports = function DBManagerModule(pb) {
>                  };
>                  cb(err, result);
>              });
> +        }
> +
> +
> +        /**
> +         * Sorts through all created indices and drops any index not declared in pb.config
> +         * @method dropUnconfiguredIndices
> +         * @param {Array} procedures
> +         * @param {Function} cb
> +         */
> +        this.dropUnconfiguredIndices = function(procedures, cb) {
> +            var self = this;
> +            //to prevent a cirular dependency we do the require for DAO here.
> +            var DAO = require('./dao.js')(pb);
> +            this.getStoredIndices(function(err, storedIndices) {


The error is not handled

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34426759>.